.github/workflows/image-builds-with-cache.yml

```diff
@@ -68,25 +70,15 @@ jobs:
           } >> "$GITHUB_OUTPUT"
 
       # Check if the image tarball already exists or not, if yes, then skip building the image
-      - name: Attempt to download the artifact
-        uses: actions/download-artifact@v4
+      - name: Check artifact exists
+        uses: ./.github/actions/check-artifact-exists
         with:
-          name: ${{ env.ARTIFACT_NAME }}
-          path: ${{ env.ARTIFACTS_PATH }}
-        continue-on-error: true
-        id: artifact-download
-
-      # If the image tarball was successfully downloaded, then clean it up as the download should
-      # happen when image tarballs are actually required for deployment (most likely in a new workflow/job)
-      - name: Delete the artifact downloaded artifact
-        if: ${{ steps.artifact-download.outcome == 'success' }}
-        run: |
-          rm -rf ${{ env.ARTIFACTS_PATH }}/${{ env.ARTIFACT_NAME }}
-        shell: bash
+          artifact_name: ${{ env.ARTIFACT_NAME }}
+        id: artifact-check
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
-        if: ${{ steps.artifact-download.outcome == 'failure' }}
+        if: ${{ steps.artifact-check.outputs.exists == 'false' }}
         id: setup-buildx
 
       - name: Build and save Docker image
```

Comment:
@nsingla  I think we also need to change the condition on this step- https://github.com/kubeflow/pipelines/blob/ff630c3ad9d9d03e50384d2165bfe6a0ec5f528b/.github/workflows/image-builds-with-cache.yml#L97
Since  steps.save-image.outcome != 'success' evaluate to true ,when` Check artifact exists` returns true and `save-image` gets skipped, so its outcome becomes skipped. But in that case, `Rebuild Images in case of failure` step should get skipped since the artifact exists.